tools/c7n_azure/c7n_azure/policy.py

```diff
@@ -331,3 +281,22 @@ def _is_subscribed_to_event(self, event, subscribed_events):
             return False
 
         return True
+
+    def _create_storage_queue(self, queue_name, session):
+        self.log.info("Creating storage queue")
+        storage_client = session.client('azure.mgmt.storage.StorageManagementClient')
+        storage_account = storage_client.storage_accounts.get_properties(
+            self.storage_account['resource_group_name'], self.storage_account['name'])
+
+        StorageUtilities.create_queue_from_storage_account(storage_account, queue_name)
+        self.log.info("Storage queue creation succeeded")
+
+        return storage_account
+
+    def _create_event_subscription(self, storage_account, queue_name, session):
+        self.log.info('Creating event grid subscription')
+        destination = StorageQueueEventSubscriptionDestination(resource_id=storage_account.id,
+                                                               queue_name=queue_name)
+
+        AzureEventSubscription.create(destination, queue_name, session)
```

Comment:
Can we log the result of calling create? If it fails, I think we need to add retry.

Reply:
Why do you think a retry would be beneficial here? We had a retry for the function webhook subscription because the function wasn't live for the handshake.

Reply:
Added try-catch-log-exit where we create the subscription.